readers/common/src/main/java/com/criteo/hadoop/garmadon/reader/GarmadonReader.java

```diff
@@ -129,18 +146,16 @@ protected void readMsg() {
                                 try {
                                     body = GarmadonSerialization.parseFrom(typeMarker, new ByteArrayInputStream(raw, bodyOffset, bodySize));
                                 } catch (DeserializationException e) {
-                                    LOGGER.debug("Cannot deserialize event from kafka record " + record + " with type " + typeMarker);
+                                    LOGGER.debug("Cannot deserialize event from kafka record {} with type {}", record, typeMarker);
+                                    break;
                                 }
                             }
-
                             CommittableOffset<String, byte[]> committableOffset = new CommittableOffset<>(consumer, record.topic(), record.partition(), record.offset());
 
-                            if (header != null && body != null) {
```

Comment:
We can leave this line for safety but still use the break you introduced to exit the filter loop